README.md

```diff
@@ -1,107 +1,207 @@
+
+```ts
+// src/shim.node.ts
+export { default as fetch } from "node-fetch";
+
+// more shims exported here
 ```
 
+Lastly, you need to register your shims in `tsconfig.json` so `deno2node` can
+inject them for you:
+
 ```jsonc
-// @filename: tsconfig.json
 {
   "deno2node": {
     "shim": "src/shim.node.ts" // path to shim file, relative to tsconfig
   }
 }
 ```
 
+If you simply want to shim all Deno globals, you can install the `deno.ns`
+package. Then do:
+
+```ts
+// src/shim.node.ts
+export * from "deno.ns";
+```
+
+That package was specifically created for this use case.
+
 ### Runtime-specific code
 
-When the provided transformations are not enough,
-you can provide a Node-specific (`<anything>.node.ts`)
-and a Deno-specific (`<anything>.deno.ts`) version of any file.
+In same cases you may want to have two different implementations, depending on
+whether you're running on Deno or on Node. When shimming is not enough, you can
+provide a Node-specific `<anything>.node.ts` and a Deno-specific
+`<anything>.deno.ts` version of any file. They need to reside next to each other
+in the same directory.
+
+`deno2node` will ignore the Deno version and rewrite imports to use the Node.js
+version instead. Thus, the Deno-specific file will not be part of the build
+output.
+
+For example, provide `greet.deno.ts` for Deno:
+
+```ts
+// src/greet.deno.ts
+export function greet() {
+  console.log("Hello Deno!");
+  // access Deno-specific APIs here
+}
+```
+
+Now, provide `greet.node.ts` for Node:
+
+```ts
+// src/greet.node.ts
+export function greet() {
+  console.log("Hello Node!");
+  // access Node-specific APIs here
+}
+```
 
-`deno2node` will ignore the Deno version
-and rewrite imports to use the Node.js version instead.
+Finally, use it in `foo.ts`:
 
-This technique has many uses.
-`deno2node` uses it to import from https://deno.land/x.
-[`grammy`] will probably also use it to abstract away platform-specific APIs.
+```ts
+import { greet } from "./platform.deno.ts";
+
+// Prints "Hello Deno!" on Deno,
+// and "Hello Node!" on Node:
+greet();
+```
+
+This technique has many uses. `deno2node` itself uses it to import from
+https://deno.land/x. A plugin of the Telegram bot framework [`grammY`] uses it
+to abstract away platform-specific APIs.
 
 ### Vendoring
 
-If you import a module which has no npm equivalent,
-`deno2node` will vendor it in `vendorDir`, if specified:
+If you import a module which has no npm equivalent, `deno2node` can extract the
+code out of Deno's module cache. On Deno, this directory can be used for
+vendoring (by specifying the `$DENO_DIR` environment variable), so `deno2node`
+calls it `vendorDir`.
 
 ```jsonc
-// @filename: tsconfig.json
+// tsconfig.json
 {
   "deno2node": {
     "vendorDir": "src/.deno2node/vendor/" // path within `rootDir`
   }
 }
 ```
 
-Vendoring requires `--allow-env`, to locate Deno cache.
+> Note that vendoring requires `--allow-env` in order to locate Deno cache.
```

Comment:
It also needs read access to Deno cache, which the example code doesn't provide. But I think I'll simply add a permission prompt.